### ISSUE TITLE
tray error tray: Failed to put tray above 0x3600001 in the stack (XCB_MATCH (8)

### ISSUE THREAD

**Issue by StarrKiss** [polybar/polybar#1801]
The tray is shifted down and to the right from the rest of the bar. All the icons working. Launching polybar through CLI shows that itgives this error: 
error: tray: Failed to put tray above 0x3600001 in the stack (XCB_MATCH (8))
every time I launch it.
![ss](https://user-images.githubusercontent.com/13007697/58961683-475e6900-876f-11e9-89a7-7ff5cf54968a.png)

Polybar config:

[polybar.txt](https://github.com/polybar/polybar/files/3257488/polybar.txt)

**Comment by patrick96** [comment]
This is because i3 repositions your bar. Set `override-redirect = true` to prevent that. Nothing else we can do about it.

**Comment by StarrKiss** [comment]
Should I place that in the polybar config or the i3 config?

**Comment by StarrKiss** [comment]
EDIT: That places the bar UNDER my windows. Is there any way to fix this?

**Comment by StarrKiss** [comment]
EDIT: EDIT: error: tray: Failed to put tray above 0x1200001 in the stack (XCB_MATCH (8))

Now I am getting this error as well

**Comment by patrick96** [comment]
> EDIT: That places the bar UNDER my windows. Is there any way to fix this?

See this: https://github.com/polybar/polybar/wiki/Known-Issues#override-redirect-in-i3

It's because i3 no longer reserves space for polybar. If you use the latest unreleased changes from i3-gaps, you can also use `gaps top` in your i3 config.

> EDIT: EDIT: error: tray: Failed to put tray above 0x1200001 in the stack (XCB_MATCH (8))

See #425, this is a known issue and we are working on a fix.

---

Alternatively, if you don't care about the bar being offset 2%, you can remove `offset-x` and `offset-y` and set `override-redirect = false`.

**Comment by averydark** [comment]
If psuedo-transparency is true, I get the error `polybar|error: tray: Failed to put tray above [MemoryAddress] in the stack (XCB_MATCH (8))`. When I set it to false, it fixes the error. 

```
[settings]

;...settings here...

; Enables pseudo-transparency for the bar
; If set to true the bar can be transparent without a compositor.
pseudo-transparency = true
```

It was working for me when I was playing around with `polybar example`. This issue popped up when I began customizing the configuration. 

Unfortunately, it was late, I was tired while I was making changes, and I have not been able to figure out was caused this. Just thought it was worth noting.

I have been playing around with the compositing operators to see if I can get the same effect, but no luck so far. 

When I have some time, I'll play around with the example config to see if it still works with the original and try to pin point what I did in the config to break it.

**Comment by averydark** [comment]
So, I got around to playing with the `example` config and narrowed down the transparency issue to the setting `border-color = #00000000` in the config file.

If this value is changed to any other value, transparency _without_ compositing seems to just fail. The error is still present where `polybar` says there's an error with the tray stack.

It still goes away when I set `pseudo-transparency` to `false`. It comes back when I set it to `true`.

Even though the error is present, it still fixes the transparency issue.

So, setting `pseudo-transparency = true` while using `border-color = #00000000` seemingly gives the desired result, but still generates this error. As far as I can tell, it seems safe to assume that the value `#00000000` is a magic number indicating transparency to something.

Hopefully this can help someone who knows more about `polybar` to narrow down the issue.
